pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -307,70 +310,59 @@ func updateEmbeddedQuota(dep rv1beta2.DeploymentInterface, obj *v3.ClusterLoggin
 	return nil
 }
 
-func getEmbeddedESEndpoint(podLister rv1.PodLister, serviceLister rv1.ServiceLister) (esEndpoint string, err error) {
+func getEndpoint(deploymentLister rv1beta2.DeploymentLister, podLister rv1.PodLister, serviceLister rv1.ServiceLister, nodeLister v3.NodeLister, clusterName, serviceName string) (esEndpoint string, err error) {
+	deploymen, err := deploymentLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
+	for _, cond := range deploymen.Status.Conditions {
+		if cond.Status == v1.ConditionFalse {
+			return "", fmt.Errorf("deployment %s status %s is %s, reason: %s, message: %s", deploymen.Name, cond.Type, cond.Status, cond.Reason, cond.Message)
+		}
+	}
+
 	selector := labels.NewSelector()
-	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{loggingconfig.EmbeddedESName})
+	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{serviceName})
 	if err != nil {
 		return "", err
 	}
 
-	espods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
+	pods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
 	if err != nil {
 		return "", err
 	}
-	esservice, err := serviceLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
+	if len(pods) == 0 {
+		return "", fmt.Errorf("deploying %s", serviceName)
+	}
+	pod := pods[0]
```

Comment:
why do we only check one of the pods? What if the pod is being recreated? Then you'd see more than one, and it's possible for only one of them being in a good state

Reply:
Also is there a reason for checking the pod condition, no the deployment? Deployment status should reflect if there is something wrong with the pod

Reply:
Also checking just one condition on the pod is not enough. Pod can be running, but not healthy (liveness/readiness probe can be failing). We should rethink the implementation - see some ideas in the comment below.